clippy_lints/src/float_literal.rs

```diff
@@ -82,20 +82,20 @@ impl<'tcx> LateLintPass<'tcx> for FloatLiteral {
                     LitFloatType::Suffixed(ast::FloatTy::F64) => Some("f64"),
                     LitFloatType::Unsuffixed => None
                 };
-                let (is_whole, mut float_str) = match fty {
+                let (is_whole, is_inf, mut float_str) = match fty {
                     FloatTy::F32 => {
                         let value = sym_str.parse::<f32>().unwrap();
 
-                        (value.fract() == 0.0, formatter.format(value))
+                        (value.fract() == 0.0, value.is_infinite(), formatter.format(value))
                     },
                     FloatTy::F64 => {
                         let value = sym_str.parse::<f64>().unwrap();
 
-                        (value.fract() == 0.0, formatter.format(value))
+                        (value.fract() == 0.0, value.is_infinite(), formatter.format(value))
                     },
                 };
 
-                if is_whole && !sym_str.contains(|c| c == 'e' || c == 'E') {
+                if is_whole && !is_inf && !sym_str.contains(|c| c == 'e' || c == 'E') {
```

Comment:
The `!is_inf` check will only be for `LOSSY_FLOAT_LITERAL` here. (verified it by reverting the change here and running uitest still passed)

We can do an early return with the inf check instead, before spanning the lint. That also makes it more readable.